packages/rocketchat-file-upload/server/lib/FileUpload.js

```diff
@@ -28,7 +31,16 @@ Object.assign(FileUpload, {
 				return `${ RocketChat.settings.get('uniqueID') }/uploads/${ file.rid }/${ file.userId }/${ file._id }`;
 			},
 			// transformWrite: FileUpload.uploadsTransformWrite
-			onValidate: FileUpload.uploadsOnValidate
+			onValidate: FileUpload.uploadsOnValidate,
+			onRead(fileId, file, req, res) {
+				if (FileUpload.requestCanAccessFiles(req)) {
```

Comment:
if you could have this `if` testing the negative condition instead the positive one (like it is done [here](https://github.com/RocketChat/Rocket.Chat/pull/9215/files#diff-d5664ca3caa8062001b0dbddf9679f7cR11)) I think it would be better =)